metrics/release.go

```diff
@@ -39,10 +39,18 @@ var (
 		[]string{},
```

Comment:
writing here, to make it easier. I am wondering it you could just keep the same metrics functions and adding the extra info as labels instead of duplicating. For instance, the`ReleaseProcessingDurationSeconds` we only need a new label saying it it is managed or tenant, or later final. That is enough to classify in the dashboard.

Reply:
We don´t track the CR name (release or pipelinerun, which I think we should add to the metrics), so we can't know if the managed and the tenant metric belongs to the same CR. Anyhow, in the current implementation, if the idea is having a metric to track specifically each pipelinerun duration, I would suggest instead of changing the ReleaseProcessingDurationSeconds (so we still can know how long the whole release took - and the current graphs does not require changes), to add a new one ie. ReleasePipelinerunsDurationSeconds that we could track all pipelines durations. - Of course, if we have had the release name and the pipelinerun name maybe only the second one would be necessary, because it would look like this:

release_pipelineruns_duration_seconds{name="release-abc", pipelinerun="tenant-abc",  type="tenant", target="...tenant", ...}

This would be easy to query and graph.

Reply:
I'm ok with adding the type but we should never add the name of the release in my opinion. It doesn't repeat over time and would create performance issues.

Reply:
Okay, I am changing it per your suggestion. I am doing it in a separate commit. If you two like how it looks, I will squash them together, but I wanted to do it separately in case I change it wrong and we want to go back to how I had it.
I am just working on fixing one thing in the test and it will be updated

Reply:
Updated

Reply:
It works for me.